tracer.go

```diff
@@ -134,6 +136,8 @@ func NewTracer(
 	} else {
 		t.logger.Error("Unable to determine this host's IP address: " + err.Error())
 	}
+	t.uuid = uuid.NewV4()
+	t.tags = append(t.tags, Tag{key: TracerUUID, value: t.uuid.String()})
```

Comment:
hm, so the reason we're adding this (btw - please create an issue first describing the problem and proposed solution) is so that debug tag can go through a rate limiter. This raises questions
  * where will that rate limiter live? should it be part of the sampler API in some way?
  * polling the agent is already done in the RemoteSampler, are we adding the polling for credits there as well?
  * the polling will need the same UUID, but the RemoteSampler is passed to the tracer externally, how will we handle that?

Reply:
- The throttler will be a wrapper around the samplers in the future so it makes sense to make it a part of the sampler API. I'll play around with where it can live. I don't want to bloat the existing jaeger namespace so I'll probably add it inside a directory and use duck typing if necessary.
- we have separate polling for baggage restrictions, I don't really see need to tie it down inside the remote sampler. Alternatively, we should extract out all this config fetching stuff in its own file because we'll have 3 separate httpProxies all hitting agent at different intervals.
- hrmmmm, it seems like the uuid needs to be generated in the config file and passed to tracer (as an option) and throttler. If people manually initialize tracer and sampler it'll be troublesome though. Does having a global uuid somewhere make sense that's lazily initialized inside the repo somewhere?

Reply:
Added #249. The UUID itself is only here because all process tags seem to be populated here. We do not plan on using the existing RemoteSampler to poll for credits AFAIK (@black-adder to confirm). The throttler should be an independent entity. I am still unsure whether or not a rate limiter is an appropriate for the client throttler, or if a simple credit balance is better. In the latter case, the throttler cannot regenerate tokens on its own, it must query the agent. The mechanism on the agent is likely to be more complex. If we follow that approach, we can even eliminate the polling and just wait for the client to exhaust its credits.

Reply:
@black-adder it should not need to be passed to the throttler because it is already a tracer/process tag.

Reply:
> it seems like the uuid needs to be generated in the config file and passed to tracer (as an option) and throttler.

I'd rather not do that. We have a similar situation with the Reporter, which is solved by having SetProcess method on the reporter. We could do the same with the debug rate limiter or sampler, depending where it's going to go.

But it seems to me we're jumping to the implementation before we actually designed how this whole thing will work in the client, can we step back and write a design that considers all aspects, not just "let's add uuid here'?